lib/modules/datasource/go/base.ts

```diff
@@ -149,6 +149,19 @@ export class BaseGoDatasource {
 
       const registryUrl = `${parsedUrl.protocol}//${parsedUrl.host}`;
 
+      // a .git path indicates a concrete git repository, which can be different from metadata returned by gitlab
+      const directUrl = URL.parse(`https://${goModule}`);
+      const vcsIndicatedModule = BaseGoDatasource.gitVcsRegexp.exec(
```

Comment:
why not pass goModule directly?

Reply:
Initially I tried to reuse the same regex for both usecases (with and without matching the gitlab URL), however, I does make more sense to adjust the regex and use it on `goModule` directly, so I adjusted the relevant code